test/python/circuit/test_singleton.py

```diff
@@ -597,3 +741,22 @@ def test_inner_outer_label_pickle(self):
         self.assertTrue(copied.mutable)
         self.assertEqual("my h gate", copied.base_gate.label)
         self.assertEqual("foo", copied.label)
+
+    def test_singleton_with_defaults(self):
+        self.assertIs(CXGate(), CXGate(label=None))
+        self.assertIs(CXGate(), CXGate(duration=None, unit="dt"))
+        self.assertIs(CXGate(), CXGate(_base_label=None))
+        self.assertIs(CXGate(), CXGate(label=None, ctrl_state=None))
+
+    def test_singleton_with_equivalent_ctrl_state(self):
+        self.assertIs(CXGate(), CXGate(ctrl_state=None))
+        self.assertIs(CXGate(), CXGate(ctrl_state=1))
+        self.assertIs(CXGate(), CXGate(label=None, ctrl_state=1))
+        self.assertIs(CXGate(), CXGate(ctrl_state="1"))
+        self.assertIsNot(CXGate(), CXGate(ctrl_state=0))
+        self.assertIsNot(CXGate(), CXGate(ctrl_state="0"))
```

Comment:
Do you want to add a test for `CXGate(ctrl_state=0) is CXGate(ctrl_state="0")`? It's covered by the discrete test cases added above, but having coverage for our stdlib usage might be good too.

Reply:
That would fail currently, because I didn't set `CXGate` to add an additional singleton that corresponds to the `ctrl_state=0`.  I can make that change (etc for other 1q controls) if you want it, though.

Reply:
Nah, it's fine it's been a long day and I just assumed it was there because the infrastructure is being added to the PR. But, that was part of my concern with this PR scope was expanding the singletons for the controlled gate control state, so it's good that it's not in the PR.

Reply:
Yeah, that's the reason I didn't do it in this PR.